modules/tasks/tasks/src/main/java/org/motechproject/tasks/util/SchedulerTaskTriggerUtil.java

```diff
@@ -0,0 +1,203 @@
+import org.motechproject.scheduler.contract.RepeatingJobId;
+import org.motechproject.scheduler.contract.RepeatingPeriodJobId;
+import org.motechproject.scheduler.contract.RepeatingPeriodSchedulableJob;
+import org.motechproject.scheduler.contract.RepeatingSchedulableJob;
+import org.motechproject.scheduler.contract.RunOnceJobId;
+import org.motechproject.scheduler.contract.RunOnceSchedulableJob;
+import org.motechproject.scheduler.exception.MotechSchedulerException;
+import org.motechproject.scheduler.service.MotechSchedulerService;
+import org.motechproject.tasks.domain.SchedulerJobType;
+import org.motechproject.tasks.domain.SchedulerTaskTriggerInformation;
+import org.motechproject.tasks.domain.Task;
+import org.motechproject.tasks.service.TaskService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This util class is responsible for the handling of scheduling, unscheduling and building
+ * scheduler jobs for Scheduler task triggers. Since the Scheduler trigger works differently to
+ * all other triggers, as it requires firing a task every specified amount of time, this util
+ * class has been created to encapsulate all the required logic in that area.
+ */
+@Component
+public class SchedulerTaskTriggerUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchedulerTaskTriggerUtil.class);
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormat.forPattern("yy-MM-dd HH:mm Z");
+
+    @Autowired
+    private MotechSchedulerService schedulerService;
+    @Autowired
+    private TaskService taskService;
+
+    public void scheduleTriggerJob(String subject) {
+        MotechEvent motechEvent = prepareSchedulerEvent(subject);
+        Task task = getSingleTaskBySubject(subject);
+
+        DateTime now = DateUtil.now();
+
+        DateTime startDate;
+        DateTime endDate;
+        boolean ignorePastFiresAtStart;
+        int interval;
+        String cronExpression;
+        List<DayOfWeek> days;
+        Time time;
+        Period repeatPeriod;
+
+        SchedulerTaskTriggerInformation trigger = (SchedulerTaskTriggerInformation) task.getTrigger();
+        SchedulerJobType triggerType = trigger.getType();
+
+        switch (triggerType) {
+            case RUN_ONCE_JOB:
+                LOGGER.info("Scheduling run once job for task: {}", task);
+                startDate = DATE_FORMAT.parseDateTime(trigger.getStartDate());
+                if (now.isBefore(startDate)) {
+                    schedulerService.safeScheduleRunOnceJob(
+                            new RunOnceSchedulableJob(motechEvent, startDate.toDate()));
+                } else {
+                    throw new MotechSchedulerException("The run once job has not been scheduled, because provided date ("
```

Comment:
Shouldn't really be the scheduler exception since this is not the scheduler. Something along the lines of IllegalArgumentException would fit here